src/Nethermind/Nethermind.Init.Snapshot/SnapshotPlugin.cs

```diff
@@ -15,8 +19,22 @@ public class SnapshotPlugin : IInitializationPlugin
     public Task InitNetworkProtocol() => Task.CompletedTask;
     public Task InitRpcModules() => Task.CompletedTask;
 
-    public bool ShouldRunSteps(INethermindApi api) =>
-        api.Config<ISnapshotConfig>() is { Enabled: true, DownloadUrl: not null };
+    public IModule? GetModule(string engineType, IConfigProvider configProvider)
+    {
+        if (configProvider.GetConfig<ISnapshotConfig>() is { Enabled: true, DownloadUrl: not null })
+        {
+            return this;
+        }
```

Comment:
Maybe like `ModuleEnabled` method first? And only if true call `GetModule`?

Reply:
Added `Enabled`.